STYLEGUIDE.md

```diff
@@ -3,25 +3,695 @@
-- [SASS](https://elastic.github.io/eui/#/guidelines/sass)
-- [HTML](style_guides/html_style_guide.md)
-- [API](style_guides/api_style_guide.md)
-- [Architecture](style_guides/architecture_style_guide.md)
-- [Accessibility](style_guides/accessibility_guide.md)
+Besides the content in this style guide also the following style guides apply
+to all development within the Kibana project. Please make sure to also read them:
 
-## Filenames
+- [Accessibility style guide](style_guides/accessibility_guide.md)
+- [SASS style guide](https://elastic.github.io/eui/#/guidelines/sass)
+
+## General
+
+### Filenames
 
 All filenames should use `snake_case`.
 
+**Right:** `src/kibana/index_patterns/index_pattern.js`
+
+**Wrong:** `src/kibana/IndexPatterns/IndexPattern.js`
+
+### Do not comment out code
+
+We use a version management system. If a line of code is no longer needed,
+remove it, don't simply comment it out.
+
+### Prettier and Linting
+
+We are gradually moving the Kibana code base over to Prettier. All TypeScript code
+and some JavaScript code (check `.eslintrc.js`) is using Prettier to format code. You
+can run `node script/eslint --fix` to fix linting issues and apply Prettier formatting.
+We recommend you to enable running ESLint via your IDE.
+
+Whenever possible we are trying to use Prettier and linting over written style guide rules.
+Consider every linting rule and every Prettier rule to be also part of our style guide
+and disable them only in exceptional cases and ideally leave a comment why they are
+disabled at that specific place.
+
+## HTML
+
+This part contains style guide rules around general (framework agnostic) HTML usage.
+
+### Camel case `id` and `data-test-subj`
+
+Use camel case for the values of attributes such as `id` and `data-test-subj` selectors.
+
+```html
+<button
+  id="veryImportantButton"
+  data-test-subj="clickMeButton"
+>
+  Click me
+</button>
+```
+
+The only exception is in cases where you're dynamically creating the value, and you need to use
+hyphens as delimiters:
+
+```jsx
+buttons.map(btn => (
+  <button
+    id={`veryImportantButton-${btn.id}`}
+    data-test-subj={`clickMeButton-${btn.id}`}
+  >
+    {btn.label}
+  </button>
+)
+```
+
+### Capitalization in HTML and CSS should always match
+
+It's important that when you write CSS/SASS selectors using classes, IDs, and attributes
+(keeping in mind that we should _never_ use IDs and attributes in our selectors), that the
+capitalization in the CSS matches that used in the HTML. HTML and CSS follow different case sensitivity rules, and we can avoid subtle gotchas by ensuring we use the
+same capitalization in both of them.
+
+## API endpoints
+
+The following style guide rules are targeting development of server side API endpoints.
+
+### Paths
+
+API routes must start with the `/api/` path segment, and should be followed by the plugin id if applicable:
+
+**Right:** `/api/marvel/nodes`
+
+**Wrong:** `/marvel/api/nodes`
+
+### snake_case
```

Comment:
should you put back the version rule, since that will be enforced at some point and is still general guidance?